test_rgb2hsv is flaky

It looks like `test_rgb2hsv` is flaky and fails sometimes.

https://app.circleci.com/pipelines/github/pytorch/vision/3217/workflows/77e60582-2ddc-46db-933f-33c45c27387c/jobs/178179/tests

Example error that we get
```python
    def test_rgb2hsv(self):
        shape = (3, 150, 100)
        for _ in range(20):
            img = torch.rand(*shape, dtype=torch.float)
            ft_hsv_img = F_t._rgb2hsv(img).permute(1, 2, 0).flatten(0, 1)
    
            r, g, b, = img.unbind(0)
            r = r.flatten().numpy()
            g = g.flatten().numpy()
            b = b.flatten().numpy()
    
            hsv = []
            for r1, g1, b1 in zip(r, g, b):
                hsv.append(colorsys.rgb_to_hsv(r1, g1, b1))
    
            colorsys_img = torch.tensor(hsv, dtype=torch.float32)
    
            max_diff = (colorsys_img - ft_hsv_img).abs().max()
>           self.assertLess(max_diff, 1e-5)
E           AssertionError: tensor(1.) not less than 1e-05

test\test_functional_tensor.py:118: AssertionError
```

--- a/test/test_functional_tensor.py
+++ b/test/test_functional_tensor.py
@@ -1,6 +1,7 @@
 import unittest
 import random
 import colorsys
+import math 
 
 from PIL import Image
 from PIL.Image import NEAREST, BILINEAR, BICUBIC
@@ -114,7 +115,13 @@ class Tester(unittest.TestCase):
 
             colorsys_img = torch.tensor(hsv, dtype=torch.float32)
 
-            max_diff = (colorsys_img - ft_hsv_img).abs().max()
+            ft_hsv_img_h, ft_hsv_img_sv = torch.split(ft_hsv_img, [1,2], dim=1)
+            colorsys_img_h, colorsys_img_sv = torch.split(colorsys_img, [1,2], dim=1)
+            
+            max_diff_h = ((colorsys_img_h*2*math.pi).sin() - (ft_hsv_img_h*2*math.pi).sin()).abs().max()
+            max_diff_sv = (colorsys_img_sv - ft_hsv_img_sv).abs().max()
+            max_diff = max(max_diff_h, max_diff_sv)
+
             self.assertLess(max_diff, 1e-5)
 
     def test_adjustments(self):

--- a/test/test_functional_tensor.py
+++ b/test/test_functional_tensor.py
@@ -1,7 +1,7 @@
 import unittest
 import random
 import colorsys
-import math 
+import math
 
 from PIL import Image
 from PIL.Image import NEAREST, BILINEAR, BICUBIC
@@ -115,10 +115,10 @@ class Tester(unittest.TestCase):
 
             colorsys_img = torch.tensor(hsv, dtype=torch.float32)
 
-            ft_hsv_img_h, ft_hsv_img_sv = torch.split(ft_hsv_img, [1,2], dim=1)
-            colorsys_img_h, colorsys_img_sv = torch.split(colorsys_img, [1,2], dim=1)
-            
-            max_diff_h = ((colorsys_img_h*2*math.pi).sin() - (ft_hsv_img_h*2*math.pi).sin()).abs().max()
+            ft_hsv_img_h, ft_hsv_img_sv = torch.split(ft_hsv_img, [1, 2], dim=1)
+            colorsys_img_h, colorsys_img_sv = torch.split(colorsys_img, [1, 2], dim=1)
+
+            max_diff_h = ((colorsys_img_h * 2 * math.pi).sin() - (ft_hsv_img_h * 2 * math.pi).sin()).abs().max()
             max_diff_sv = (colorsys_img_sv - ft_hsv_img_sv).abs().max()
             max_diff = max(max_diff_h, max_diff_sv)
 

